packages/wrangler/src/dev/local.tsx

```diff
@@ -236,43 +236,45 @@ function useLocalWorker({
 				__dirname,
 				"../miniflare-dist/index.mjs"
 			);
-			const optionsArg = JSON.stringify(options, null);
+			const miniflareOptions = JSON.stringify(options, null);
 
 			logger.log("⎔ Starting a local server...");
-			const localServerOptions = [
+			const nodeOptions = [
 				"--experimental-vm-modules", // ensures that Miniflare can run ESM Workers
 				"--no-warnings", // hide annoying Node warnings
-				miniflareCLIPath,
-				optionsArg,
 				// "--log=VERBOSE", // uncomment this to Miniflare to log "everything"!
 			];
 			if (inspect) {
-				localServerOptions.push("--inspect"); // start Miniflare listening for a debugger to attach
+				nodeOptions.push("--inspect"); // start Miniflare listening for a debugger to attach
 			}
-			// spawn isn't technically synchronous here
-			local.current = spawn("node", localServerOptions, {
-				cwd: path.dirname(scriptPath),
+			const child = (local.current = fork(
+				miniflareCLIPath,
+				[miniflareOptions],
+				{
+					cwd: path.dirname(scriptPath),
+					execArgv: nodeOptions,
+				}
+			));
+			child.on("message", (message) => {
+				if (message === "ready") {
+					onReady?.(local?.current?.pid);
```

Comment:
Can we use child here too?

```suggestion
					onReady?.(child.pid);
```